pkg/manager/config/config.go

```diff
@@ -976,11 +975,23 @@ func (i *Instance) setDefaultValues() error {
 	// Set default scrapers and plugins paths
 	viper.SetDefault(ScrapersPath, defaultScrapersPath)
 	viper.SetDefault(PluginsPath, defaultPluginsPath)
-	return viper.WriteConfig()
+	if write {
+		return viper.WriteConfig()
+	}
+
+	return nil
 }
 
-// SetInitialConfig fills in missing required config fields
 func (i *Instance) SetInitialConfig() error {
+	return i.setInitialConfig(true)
+}
+
+func (i *Instance) SetInitialMemoryConfig() error {
+	return i.setInitialConfig(false)
+}
+
+// SetInitialConfig fills in missing required config fields
```

Comment:
This comment should be moved back to its original position above the public method.

Reply:
Oh, darn. I'll push a fix.